homeassistant/components/switch/child_switch.py

```diff
@@ -0,0 +1,252 @@
+
+-callback_service
+The name of the service the child will call when turned on or off
+
+-callback_event
+The event that will be fired when the child is successfully created
+
+-listen_event
+This is the name of the event that the child will listen to for state
+changes from the parent
+
+
+STEP 3: Create a service for the turn on and turn off events
+In the setup function of your component define the service that will
+be called when your switch is toggled, eg:
+
+def handle_motion_detection_service(service):
+
+    target_cameras = component.extract_from_service(service)
+
+    for camera in target_cameras:
+        if action == SWITCH_ACTION_RECORD:
+            if (state == STATE_ON and not
+                    camera.is_recording):
+                camera.record_stream()
+            elif (state == STATE_OFF and
+                    camera.is_recording):
+                camera.stop_recording()
+
+        camera.update_ha_state(True)
+
+hass.services.register(
+    DOMAIN,
+    SERVICE_CAMERA,
+    handle_motion_detection_service)
+
+
+STEP 4: Listen for child creation
+Inside you entity class you need to set up listeners for the
+child creation events.  The callback event will fire on the
+successfull creation of the child entity, eg:
+
+def add_child_component_listeners(self):
+    self.hass.bus.listen(
+        self.entity_id + EVENT_CALLBACK_RECORD,
+        self.process_switch_creation)
+
+STEP 5: Notify the child switch of parent state changes
+Add a function to your entity that fires an event for a particular
+state change.  You can then call this function whenever you want to
+notify the child switches state, eg:
+
+def send_recording_state(self):
+    state = STATE_OFF
+    if self.is_recording:
+        state = STATE_ON
+
+    self.hass.bus.fire(
+        self.entity_id + EVENT_CHANGE_RECORD,
+        {
+            'entity_id': self.entity_id,
+            'state': state
+        })
+
+If everything goes according to plan you will now see the additonal
+switch on the UI.  For a complete working example check out the camera
+component's __init__.py
+
+"""
+import logging
+
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.loader import get_component
+
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, ATTR_ENTITY_ID,
+    ATTR_DOMAIN)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the GenericSwitch platform. """
+    logger = logging.getLogger(__name__)
+
+    if discovery_info is not None:
+        parent_domain = discovery_info.get(ATTR_DOMAIN, None)
+        parent_component = get_component(parent_domain)
+
+        if parent_component is None:
+            logger.error(
+                'Could not find parent component %s', parent_domain)
+            return
+
+        parent_entity_id = discovery_info.get(ATTR_ENTITY_ID, None)
+        name = discovery_info.get('name', parent_entity_id + ' Switch')
```

Comment:
It seems that name is already specified in the discovery data (at least in the camera).